src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int capacity = DEFAULT_CAPACITY;
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int indexOfBucket = getIndexOfBucketByKey(key);
+        if (table[indexOfBucket] == null) {
+            table[indexOfBucket] = new Node<>(key, value);
+            size++;
+        } else {
+            putIfNodeInBucketExist(key, value, indexOfBucket);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexOfBucket = getIndexOfBucketByKey(key);
+        if (table[indexOfBucket] != null && table[indexOfBucket].next == null) {
+            return table[indexOfBucket].value;
+        } else {
```

Comment:
Why do you need this IF at all? ))

Reply:
Sure, it's unnecessary, it just piles up code, it should be removed)